source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -636,6 +664,32 @@ namespace
         }
     }
 
+    void writeMaterialElements(PvtNode* node, NodePtr mxNode, const string& nodeName, DocumentPtr doc, const RtWriteOptions* writeOptions)
```

Comment:
Calls the writeMaterialElementsHelper with for each material node if ADD_MATERIAL_NODES_FOR_SHADERS is set with the material node name, otherwise, calls the writeMaterialElementsHelper with the name of the surface shader to create the material element from.